server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/MailboxSet.scala

```diff
@@ -0,0 +1,62 @@
+package org.apache.james.jmap.mail
+
+import eu.timepit.refined.api.Refined
+import eu.timepit.refined.collection.NonEmpty
+import org.apache.james.jmap.mail.MailboxName.MailboxName
+import org.apache.james.jmap.mail.MailboxSetRequest.MailboxCreationId
+import org.apache.james.jmap.model.AccountId
+import org.apache.james.jmap.model.State.State
+import org.apache.james.mailbox.Role
+import org.apache.james.mailbox.model.MailboxId
+import play.api.libs.json.JsObject
+
+case class MailboxSetRequest(accountId: AccountId,
+                             ifInState: Option[State],
+                             create: Option[Map[MailboxCreationId, JsObject]],
+                             update: Option[Map[MailboxId, MailboxPatchObject]],
+                             destroy: Option[Seq[MailboxId]],
+                             onDestroyRemoveEmails: Option[RemoveEmailsOnDestroy]
+                            )
+
+
+object MailboxSetRequest {
+  type MailboxCreationId = String Refined NonEmpty
+}
+
+case class RemoveEmailsOnDestroy(value: Boolean) extends AnyVal
+//TODO the parentId could be a backreference to a MailboxCreationId
+//TODO should we handle server-set parameters which should be allowed in request and that we should match to be the same than the response?
+case class MailboxCreationRequest(name: MailboxName, parentId: Option[MailboxId])
+
+
+//TODO strongly type when implementing update
+case class MailboxPatchObject(value: Map[String, JsObject])
+
+case class MailboxSetResponse(accountId: AccountId,
+                              oldState: Option[State],
+                              newState: State,
+                              created: Option[Map[MailboxCreationId, MailboxCreationResponse]],
+                              updated: Option[Map[MailboxId, MailboxUpdateResponse]],//TODO could be a backreference to a MailboxCreationId
+                              destroyed: Option[Seq[MailboxId]],//TODO could be a backreference to a MailboxCreationId,
```

Comment:
Yes. you need a '#' before your id to say that it is a backreference to a client temporary id (MailboxCreationId).
Is it right ?

We added this TODO because we thought it is possible in the same set mailbox request.
To create a mailbox
Delete it afterward.

But maybe it is not the case ? However i can remove the todo as like you said all the next steps have already a dedicated isssue.

Reply:
That's not what I meant.

MailboxCreationId is just an id set by the client.

Consider a requests that creates a mailbox then happen a message to it.

```
 A map of a creation id (a temporary id set by the client) to Foo objects, 

Some records may hold references to other records (foreign keys). That reference may be set (via create or update) in the same request as the referenced record is created. To do this, the client refers to the new record using its creation id prefixed with a #.
```

It allows referencing entities YOU create.

It is yet different from `ResultReference` defined earlier in the spec, who allow you to reference objects previously returned by method calls.